Sprint 6/step 1

Ekaterina64 opened the pull request.

Ekaterina-ux committed:
Jest тесты редьюсеров

Ekaterina-ux committed:
Cypress тесты конструктора

Ekaterina-ux committed:
Фикс cypress тестов

Ekaterina-ux committed:
Установлен лого и пакеты

Ekaterina-ux committed:
Script

Ekaterina-ux committed:
Update README

gennady-bars commented on src/services/reducers/burger-constructor/burger-constructor.test.js:
```diff
@@ -0,0 +1,184 @@
+
+	it('should set a orderRequest flag', () => {
+		expect(constructorReducer(initialState, actions.getOrderReq())).toEqual({
+			...initialState,
```
`			...initialState,
`

Хорошее использование `spread`

gennady-bars commented on cypress/e2e/constructor.cy.ts:
```diff
@@ -0,0 +1,88 @@
+			fixture: 'order.json',
+		}).as('postOrder')
+
+		cy.visit('http://localhost:3000')
```
`		cy.visit('http://localhost:3000')
`

`http://localhost:3000` будет повторяться несколько раз в коде тестов. Нужно вынести в константу `testUrl`, чтобы можно было одним движением изменить урл для тестов
А можно в документации посмотреть, как настроить быстро `baseUrl` для cypress https://docs.cypress.io/api/commands/visit#Visit-is-automatically-prefixed-with-baseUrl

gennady-bars commented on cypress/e2e/constructor.cy.ts:
```diff
@@ -0,0 +1,88 @@
+		cy.get(dataTestID('button_submit')).click()
+		cy.wait('@postOrder')
+		cy.get(dataTestID('modal'))
+			.as('modal')
```
`.as('modal')`

* Отлично, что знаете про `alias` (переменные с `@`)

gennady-bars commented on cypress/e2e/constructor.cy.ts:
```diff
@@ -0,0 +1,88 @@
+	})
+
+	it('should drag and drop a bun ingredient into constructor', () => {
+		cy.get(dataTestID('burger_ingredient')).first().trigger('dragstart')
```
`'burger_ingredient'` 3 раза дублируется в тестах

Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора.

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:


* Приложение задеплоено
* Все юнит-тесты (27 шт) проходят успешно
* Все cypress-тесты проходят успешно
* Отлично, что знаете про `alias` (переменные с `@`)
* Код тестов аккуратный и понятный


но есть некоторые недочеты:


* `http://localhost:3000` будет повторяться несколько раз в коде тестов. Нужно вынести в константу `testUrl`, чтобы можно было одним движением изменить урл для тестов
А можно в документации посмотреть, как настроить быстро `baseUrl` для cypress https://docs.cypress.io/api/commands/visit#Visit-is-automatically-prefixed-with-baseUrl 
* Неавторизованный пользователь **не должен** отправлять заказ на сервер. При клике на кнопку его нужно переводить на логин и после удачного логина обратно к заказу.
* Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора.

Ekaterina-ux committed:
Исправлено:
- Не авторизированный пользователь не может делать заказ
- Строчной селектор в константе
- url для тестов

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачи в Вашей новой профессии.

* в `README.md` теперь нужно **добавить описание проекта, технологий**, как скачать и запустить, как задеплоить и **ссылку на задеплоенный сайт**. 

**Несколько советов для дальнейшего развития:**

* сверстайте несколько небольших проектов для себя, для закрепления знаний, и чтобы можно добавить их в портфолио
* ходите на собеседования: опыт прохождения собеседований не менее важен знаний теории, и Вы поймете, какие знания требуются на рынке, и где Вы проседаете
* читайте [habr](https://habr.com/ru/) и слушаете подкаксты: они очень хорошо расширяют кругозор и позволяют узнать, что происходит в отрасли:

https://www.youtube.com/channel/UCZeU17nbVfzczAkJVTay9vw
(ссылки на каналы могут быть недоступными сейчас. Попробуйте через `vpn`)

https://soundcloud.com/everyonecan  

https://soundcloud.com/podlodka

https://soundcloud.com/devschacht

https://soundcloud.com/begebot

https://soundcloud.com/web-standards
